src/mmw/apps/modeling/tests.py

```diff
@@ -45,6 +77,75 @@ def test_tr55_job_runs_in_chain(self):
                          'complete',
                          'Job found but incomplete.')
 
+    @override_settings(CELERY_ALWAYS_EAGER=True)
+    def test_tr55_chain_skips_census_if_census_is_up_to_date(self):
+        # Census with current modification_hash
+        self.model_input['census'] = {
+            'cell_count': 100,
+            'distribution': {
+                'c:commercial': {
+                    'cell_count': 70
+                },
+                'a:deciduous_forest': {
+                    'cell_count': 30
+                }
+            },
+            'modification_hash': '39f488abec2de49f17652631ae843946',
+            'modifications': [{
+                'bmp': 'no_till',
+                'cell_count': 1,
+                'distribution': {
+                    'a:deciduous_forest': {'cell_count': 1},
+                }
+            }]
+        }
+
+        task_list = views._initiate_tr55_job_chain(self.model_input,
+                                                   self.job.id)
+
+        task_count = len(list(task_list._parents()))
+        self.assertEqual(task_count, 1,
+                         'Task list contains the wrong number of tasks')
+
+    @override_settings(CELERY_ALWAYS_EAGER=True)
+    def test_tr55_chain_generates_census_if_census_is_stale(self):
+        # Census with stale modification_hash
+        self.model_input['census'] = {
+            'cell_count': 100,
+            'distribution': {
+                'c:commercial': {
+                    'cell_count': 70
+                },
+                'a:deciduous_forest': {
+                    'cell_count': 30
+                }
+            },
+            'modification_hash': 'j3jk3jk3jn3knm3nmn39usd',
+            'modifications': [{
+                'bmp': 'no_till',
+                'cell_count': 1,
+                'distribution': {
+                    'a:deciduous_forest': {'cell_count': 1},
+                }
+            }]
+        }
+
+        task_list = views._initiate_tr55_job_chain(self.model_input,
+                                                   self.job.id)
+
+        task_count = len(list(task_list._parents()))
+        self.assertEqual(task_count, 2,
+                         'Task list contains the wrong number of tasks')
+
+    @override_settings(CELERY_ALWAYS_EAGER=True)
+    def test_tr55_chain_generates_census_if_census_does_not_exist(self):
+        task_list = views._initiate_tr55_job_chain(self.model_input,
+                                                   self.job.id)
+
+        task_count = len(list(task_list._parents()))
+        self.assertEqual(task_count, 2,
```

Comment:
These tests aren't the greatest (ideally, we would inspect which tasks get run), but I spent a few hours trying to figure out how to do that and came up empty.